src/modules/LoggedIn/Project/ProjectForm.tsx

```diff
@@ -1,16 +1,27 @@
 import React, { ChangeEvent } from "react";
 import { Button, Modal } from "antd";
-import { useProjectSync, ProjectRequest } from "./useProjectSync";
-const ProjectForm = () => {
+
+import { useProjectSync, ProjectRequest, Project } from "./useProjectSync";
+
+type FormProps = {
+  visible: boolean;
+  toggleModal: () => void;
+  project?: Project;
+};
+
+const ProjectForm = (props: FormProps) => {
   const projectSync = useProjectSync();
-  const [visible, setVisible] = React.useState<boolean>(false);
   const [newData, setNewData] = React.useState<ProjectRequest>({
     name: "",
     description: "",
   });
-  const toggleModal = React.useCallback(() => {
-    setVisible(prev => !prev);
-  }, []);
+
+  React.useEffect(() => {
+    setNewData({
+      name: props.project ? props.project.name : "",
+      description: props.project ? props.project.description : "",
+    });
+  }, [props]);
```

Comment:
harusnya dependency nya cukup `props.project` aja

```suggestion
  }, [props.project]);
```